rules/xcodeproj.bzl

```diff
@@ -785,6 +858,38 @@ Tags for configuration:
     xcodeproj-ignore-as-target: Add this to a rule declaration so that this rule will not generates a scheme for this target
 """,
     attrs = {
+        "build_configurations_debug": attr.string_list(
+            mandatory = False,
+            default = [],
+            doc = "List of names for build configurations of type 'debug' to generate. If 'Debug' is not present it will be added.",
+        ),
+        "build_configurations_debug_bazelrc": attr.string_dict(
```

Comment:
Could we directly use the build_configurations_debug/build_configurations_release values in build_wrapper and get rid of build_configurations_debug_bazelrc and build_configurations_release_bazelrc?

Reply:
This would force people to create configs for all specified configurations in the `.bazelrc` file, including `Debug` for example that, from a user perspective, should 'just work'.

Unless there's another way to conditionally append `--config <config name>` to the invocation that happens in the build wrapper?

Reply:
Yeah, I raised the concern in the original [PR](https://github.com/bazel-ios/rules_ios/pull/171). Here is @jerrymarino 's comment:
```
We might need to do a bit of research to determine how a default config could that'd fit into the picture. There might be new features that allow this ( I'm not sure ). In the worst case we could always just require a config to be present for each possibility.
```
And I didn't find any better way with the default config.

Reply:
@jerrymarino thoughts here? I feel like it's better to add this as an optional thing (if you want to use what is in your `.bazelrc` then just set `build_configurations_debug/release_bazelrc`, otherwise the rule just creates the build configuration as a "pure" 'debug'/'release' one)

Reply:
I'd consider using an API that's as minimal as possible and ideally the user could provide all of this data as a mapping:
```
{
    "Debug": "--config my_debug",
    "ReleaseEnterprise" : "--config my_release_enterprise",
    "ReleaseAppstore" : "--config my_release_appstore",
}
```

Then inside the rule add `Debug=""` and `Release=""` if they're empty.

Reply:
Perhaps there's a simpler API - What do you think about a mapping where the user can provide configs they'd like
```
 {
   Basic: "--config my_cofig",
   AppStore: "--config appstore"
}
```
Then insert `Debug = ""` and `Release = ""` if it's required by XcodeGen.

Reply:
My only concern with accepting any string here `Basic: "--config my_cofig",` is that this will ultimately being passed to the `build-wrapper.sh` so anything in this string will be passed as a `$BAZEL_RULES_IOS_OPTIONS` in there.

This means that this attribute is really `build_options_by_configuration` or something like that right? We're not just specifying the build configurations that should be created. 

I used your idea [here](https://github.com/bazel-ios/rules_ios/pull/171#pullrequestreview-550484114) as a reference, that's why I went in this path